Lib/fontTools/ttLib/ttFont.py

```diff
@@ -381,11 +381,13 @@ def keys(self):
 		keys = sortedTagList(keys)
 		return ["GlyphOrder"] + keys
 
-	def ensureDecompiled(self):
+	def ensureDecompiled(self, recurse=None):
 		"""Decompile all the tables, even if a TTFont was opened in 'lazy' mode."""
 		for tag in self.keys():
 			table = self[tag]
-			if self.lazy is not False and hasattr(table, "ensureDecompiled"):
+			if recurse is None:
+				recurse = self.lazy is not False
+			if recurse and hasattr(table, "ensureDecompiled"):
 				table.ensureDecompiled()
```

Comment:
table.ensureDecompiled also has a `recurse=True` parameter, maybe you wanted to pass that on when you call it? i.e. `table.ensureDecompiled(recurse=recurse)`.

https://github.com/fonttools/fonttools/blob/9716b1c4556642f92d66d8ceee0225e6c5ade8bb/Lib/fontTools/ttLib/tables/otBase.py#L227-L228

It doesn't seem like you're calling `TTFont.ensureDecompiled` method with a `recurse` value other than `None`. I wonder it's actually needed.

Reply:
cmap's ensureDecompiled doesn't take recurse, but yeah I think I'll fix that one too.

This change ended up being unused but I kept it. I'll probably put back recurse=False to the visitor.